src/Aspire.Hosting.Azure/AzureBicepResource.cs

```diff
@@ -219,6 +242,176 @@ public virtual void WriteToManifest(ManifestPublishingContext context)
         }
     }
 
+    /// <summary>
+    /// Provisions this Azure Bicep resource using the bicep provisioner.
+    /// </summary>
+    /// <param name="context">The pipeline step context.</param>
+    /// <param name="resource">The Azure Bicep resource to provision.</param>
+    private static async Task ProvisionAzureBicepResourceAsync(PipelineStepContext context, AzureBicepResource resource)
+    {
+        // Skip if the resource is excluded from publish
+        if (resource.IsExcludedFromPublish())
+        {
+            return;
+        }
+
+        // Skip if already provisioned
+        if (resource.ProvisioningTaskCompletionSource != null &&
+            resource.ProvisioningTaskCompletionSource.Task.IsCompleted)
+        {
+            return;
+        }
+
+        var bicepProvisioner = context.Services.GetRequiredService<IBicepProvisioner>();
+        var configuration = context.Services.GetRequiredService<IConfiguration>();
+        
+        // Find the AzureEnvironmentResource from the application model
+        var azureEnvironment = context.Model.Resources.OfType<AzureEnvironmentResource>().FirstOrDefault();
+        if (azureEnvironment?.ProvisioningContext == null)
+        {
+            throw new InvalidOperationException("AzureEnvironmentResource with a valid ProvisioningContext must be present in the application model.");
+        }
+        
+        var provisioningContext = azureEnvironment.ProvisioningContext;
+
+        var resourceTask = await context.ReportingStep
+            .CreateTaskAsync($"Deploying **{resource.Name}**", context.CancellationToken)
+            .ConfigureAwait(false);
+
+        await using (resourceTask.ConfigureAwait(false))
+        {
+            try
+            {
+                resource.ProvisioningTaskCompletionSource =
```

Comment:
Initialize the ProvisioningTaskCompletionSource for this resource during the creation of step not in the execution callback.